angular-workspace/projects/ni/nimble-angular/pipes/format-number-decimal.pipe.ts

```diff
@@ -0,0 +1,64 @@
+import { Inject, LOCALE_ID, Pipe, PipeTransform } from '@angular/core';
+import { DecimalUnitFormat } from '@ni/nimble-components/dist/esm/utilities/unit-format/decimal-unit-format';
+import { TableColumnNumberText } from '@ni/nimble-components/dist/esm/table-column/number-text';
+import { passthroughUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/passthrough-unit-scale';
+import { byteUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/byte-unit-scale';
+import { byte1024UnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/byte-1024-unit-scale';
+import { voltUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/volt-unit-scale';
+import type { UnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/unit-scale';
+
+export { byteUnitScale, byte1024UnitScale, voltUnitScale };
+
+/**
+ * A pipe that transforms a decimal number into a string with optional unit, formatted like "1.2 kB"
+ */
+@Pipe({
+    name: 'formatNumberDecimal',
+    standalone: true
+})
+export class FormatNumberDecimalPipe implements PipeTransform {
+    /**
+     * @internal
+     */
+    public decimalUnitFormat?: DecimalUnitFormat;
+    private minimumFractionDigits?: number;
+    private maximumFractionDigits?: number;
+    private unitScale?: UnitScale;
+
+    public constructor(@Inject(LOCALE_ID) private readonly locale: string) {}
+
+    public transform(value: number, {
+        decimalDigits,
+        maximumDecimalDigits,
+        unitScale = passthroughUnitScale
+    }: {
+        decimalDigits?: number,
+        maximumDecimalDigits?: undefined,
+        unitScale?: UnitScale
+    } | {
+        decimalDigits?: undefined,
+        maximumDecimalDigits?: number,
+        unitScale?: UnitScale
+    } = {
+        unitScale: passthroughUnitScale
+    }): string {
+        const { minimumFractionDigits, maximumFractionDigits } = TableColumnNumberText.convertDigitOptionsForDecimalUnitFormat(
```

Comment:
@m-akinc @mollykreis we're now in a weird cross where the api in angular does not represent any api in components. 

We should pick either:
- Expose the UnitFormat api directly as pipes (and the future goal of both the vanilla js and angular pipe moving to separate libraries)
- Expose an API from components that encapsulates the details of the table api.

Seems like work was going in here to align closer to the existing table number column api vs the separate unit format api (by overriding the default value behavior of the unit format apis). If that's the intended direction lets go all the way.

Proposal:

Create: `nimble-components/src/table-column/number-text/models/number-text-unit-format.ts`

```ts
class NumberTextUnitFormat extends UnitFormat {
   constructor(locale: string, options: {
         numberTextFormat: NumberTextFormat,
         decimalDigits?: number, // comment that unused for certain formats
         maximumDecimalDigits?: undefined, // comment that unused for certain formats
         unitScale?: UnitScale;
   }
   ) { /* encapsulate all the table logic of default values */}
}
```

And then we can use that in the table number column (it won't need to reference the separate unitformat pipes directly anymore) and then make a single `NumberTextUnitFormatPipe`. Thoughts?

Reply:
Makes sense. Any objection to making a factory rather than an actual, new subclass of `UnitFormat`? (I hit a snag trying to pass-through a call to `tryFormat`, which is `protected`.)

Reply:
It's not clear to me what the problem is from your description. You would implement the tryFormat interface and that would involve calls to the instances of DecimalUnitFormat.format and DefauultUnitFormat.format. I would not expect you to need to access their private tryFormat methods like the Angular Pipe usage didn't need to.

Reply:
The comment was resolved but didn't go the direction with the equals statement. Think this would have been an instance that would have been helpful to discuss why that suggestion was not followed and resolved without discussion.

I pushed changes that go in that direction with a `optionsMatch` method. That with a type for the options moved effectively all logic out of the Pipe. Re-opened this thread to use for discussion of that implementation. Definitely lets discuss any typing or approach Qs potentially offline if that's more efficient.

Reply:
Whoops, I think I initially didn't understand where I would need an `equals` implementation, and then I forgot about it when I switched from my factory to a new `UnitFormat` subclass. You're not saying we still need `equals`, though, right? You had just meant for me to encapsulate the configuration equality check like you ended up doing in `optionsMatch`?

Reply:
yep

Reply:
Your `optionsMatch` approach and implementation look good to me. Resolving since I don't think there are any further questions to discuss.